win32ss/user/user32/windows/cursoricon.c

```diff
@@ -2622,15 +2859,29 @@ HICON WINAPI CreateIconFromResourceEx(
             pbIconBits = (PBYTE)pt;
         }
 
-        if (!CURSORICON_GetCursorDataFromBMI(&cursorData, (BITMAPINFO*)pbIconBits))
+        /* Try to load BMP icon */
+        if (!CURSORICON_GetCursorDataFromBMI(&cursorData, (PBITMAPINFO)pbIconBits))
         {
-            ERR("Couldn't fill the CURSORDATA structure.\n");
-            if (ResHandle)
-                FreeResource(ResHandle);
-            return NULL;
+            /* Convert PNG raw data to BMP icon if the icon was PNG icon */
+            pbBmpIcon = CURSORICON_ConvertPngToBmpIcon(pbIconBits, cbIconBits, &BmpIconSize);
+            if (!pbBmpIcon)
+                goto end; /* Not PNG icon or failed */
+
+            /* Find icon entry from BMP icon */
+            CURSORICONFILEDIR *dir = (CURSORICONFILEDIR *)pbBmpIcon;
+            CURSORICONFILEDIRENTRY *entry = &dir->idEntries[0]; /* Only one entry */
+
+            /* A bit of preparation */
+            RtlZeroMemory(&cursorData, sizeof(cursorData));
+            cursorData.rt = LOWORD(fIcon ? RT_ICON : RT_CURSOR);
```

Comment:
I don't know whether it's intended here, but by ZeroMemory-ing cursorData, you erase what has been set for the xHotspot and yHotspot at line https://github.com/reactos/reactos/pull/7709/files#diff-0bde9244bc3fdf95faed2fc129956a4ce8e298dd37f39d9ad67030e7988d6bf6R2858 .
(And you are forced to redo the cursorData.rt = .... with the same code.)